flink-core/src/test/java/org/apache/flink/core/classloading/ComponentClassLoaderTest.java

```diff
@@ -0,0 +1,264 @@
+    // Resource loading
+    // ----------------------------------------------------------------------------------------------
+
+    @Test
+    public void testComponentOnlyIsDefaultForResources() throws IOException {
+        TestUrlClassLoader owner = new TestUrlClassLoader();
+
+        final ComponentClassLoader componentClassLoader =
+                new ComponentClassLoader(new URL[0], owner, new String[0], new String[0]);
+
+        assertThat(componentClassLoader.getResource(NON_EXISTENT_RESOURCE_NAME), nullValue());
+        assertThat(
+                componentClassLoader.getResources(NON_EXISTENT_RESOURCE_NAME).hasMoreElements(),
+                is(false));
+    }
+
+    @Test
+    public void testOwnerFirstResourceFoundIgnoresComponent() {
+        TestUrlClassLoader owner =
+                new TestUrlClassLoader(resourceToLoad, RESOURCE_RETURNED_BY_OWNER);
+
+        final ComponentClassLoader componentClassLoader =
+                new ComponentClassLoader(
+                        new URL[] {}, owner, new String[] {resourceToLoad}, new String[0]);
+
+        final URL loadedResource = componentClassLoader.getResource(resourceToLoad);
+        assertThat(loadedResource, sameInstance(RESOURCE_RETURNED_BY_OWNER));
+    }
+
+    @Test
+    public void testOwnerFirstResourceNotFoundFallsBackToComponent() throws Exception {
+        TestUrlClassLoader owner = new TestUrlClassLoader();
+
+        final ComponentClassLoader componentClassLoader =
+                new ComponentClassLoader(
+                        new URL[] {TMP.getRoot().toURI().toURL()},
+                        owner,
+                        new String[] {resourceToLoad},
+                        new String[0]);
+
+        final URL loadedResource = componentClassLoader.getResource(resourceToLoad);
+        assertThat(loadedResource.toString(), containsString(resourceToLoad));
+    }
+
+    @Test
+    public void testComponentFirstResourceFoundIgnoresOwner() throws Exception {
+        TestUrlClassLoader owner =
+                new TestUrlClassLoader(resourceToLoad, RESOURCE_RETURNED_BY_OWNER);
+
+        final ComponentClassLoader componentClassLoader =
+                new ComponentClassLoader(
+                        new URL[] {TMP.getRoot().toURI().toURL()},
+                        owner,
+                        new String[0],
+                        new String[] {resourceToLoad});
+
+        final URL loadedResource = componentClassLoader.getResource(resourceToLoad);
+        assertThat(loadedResource.toString(), containsString(resourceToLoad));
+    }
+
+    @Test
+    public void testComponentFirstResourceNotFoundFallsBackToOwner() {
+        TestUrlClassLoader owner =
+                new TestUrlClassLoader(NON_EXISTENT_RESOURCE_NAME, RESOURCE_RETURNED_BY_OWNER);
+
+        final ComponentClassLoader componentClassLoader =
+                new ComponentClassLoader(new URL[0], owner, new String[0], new String[] {"foo"});
```

Comment:
Why is it here `{"foo"}` and not `{resourceToLoad}`?

Reply:
It should be `NON_EXISTENT_RESOURCE_NAME`, leftover from a prior version.